app/javascript/menu/menu-collapse.jsx

```diff
@@ -6,8 +6,13 @@ import { SideNavItem } from 'carbon-components-react/es/components/UIShell';
 const MenuCollapse = ({ expanded, toggle }) => (
   <SideNavItem className="menu-collapse">
     <div
+      role="button"
+      tabIndex="0"
       className="menu-collapse-button"
       onClick={toggle}
+      onKeyPress={toggle}
+      aria-expanded={expanded}
+      aria-controls="main-menu-primary"
```

Comment:
i think this div should have aria-haspopup="true" as well, check here https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html